[2025-05-02T10:52:52.382+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-05-02T10:52:52.416+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: gcs_silver_to_golden_bigquery.transform_and_upload_golden manual__2025-05-02T10:52:33.516872+00:00 [queued]>
[2025-05-02T10:52:52.433+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: gcs_silver_to_golden_bigquery.transform_and_upload_golden manual__2025-05-02T10:52:33.516872+00:00 [queued]>
[2025-05-02T10:52:52.434+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-05-02T10:52:52.539+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): transform_and_upload_golden> on 2025-05-02 10:52:33.516872+00:00
[2025-05-02T10:52:52.548+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=2306) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-05-02T10:52:52.551+0000] {standard_task_runner.py:63} INFO - Started process 2308 to run task
[2025-05-02T10:52:52.551+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'gcs_silver_to_golden_bigquery', 'transform_and_upload_golden', 'manual__2025-05-02T10:52:33.516872+00:00', '--job-id', '60', '--raw', '--subdir', 'DAGS_FOLDER/gcs_silver_to_gold_layer_and_bigquery.py', '--cfg-path', '/tmp/tmp4awla28z']
[2025-05-02T10:52:52.556+0000] {standard_task_runner.py:91} INFO - Job 60: Subtask transform_and_upload_golden
[2025-05-02T10:52:52.663+0000] {task_command.py:426} INFO - Running <TaskInstance: gcs_silver_to_golden_bigquery.transform_and_upload_golden manual__2025-05-02T10:52:33.516872+00:00 [running]> on host dfbad22d9104
[2025-05-02T10:52:53.589+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='gcs_silver_to_golden_bigquery' AIRFLOW_CTX_TASK_ID='transform_and_upload_golden' AIRFLOW_CTX_EXECUTION_DATE='2025-05-02T10:52:33.516872+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-02T10:52:33.516872+00:00'
[2025-05-02T10:52:53.594+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-05-02T10:52:53.604+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/utils/context.py:320: AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
  warnings.warn(_create_deprecation_warning(k, replacements))

[2025-05-02T10:52:53.623+0000] {gcs_silver_to_gold_layer_and_bigquery.py:25} INFO - Starting transformation at 20250502_105233
[2025-05-02T10:52:53.629+0000] {gcs_silver_to_gold_layer_and_bigquery.py:29} INFO - Created temporary directory: /tmp/tmp9ns70jfj
[2025-05-02T10:52:53.670+0000] {connection.py:274} WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.
[2025-05-02T10:52:53.687+0000] {base.py:84} INFO - Using connection ID 'google_cloud_default' for task execution.
[2025-05-02T10:52:56.614+0000] {gcs_silver_to_gold_layer_and_bigquery.py:38} INFO - Found files in silver layer: ['silver_layer/transformed_data_20250502_094703.csv']
[2025-05-02T10:52:56.626+0000] {gcs_silver_to_gold_layer_and_bigquery.py:55} INFO - Selected latest file: silver_layer/transformed_data_20250502_094703.csv
[2025-05-02T10:52:59.700+0000] {gcs.py:347} INFO - File downloaded to /tmp/tmp9ns70jfj/transformed_data_20250502_094703.csv
[2025-05-02T10:52:59.704+0000] {gcs_silver_to_gold_layer_and_bigquery.py:64} INFO - Downloaded silver_layer/transformed_data_20250502_094703.csv to /tmp/tmp9ns70jfj/transformed_data_20250502_094703.csv
[2025-05-02T10:53:00.940+0000] {gcs_silver_to_gold_layer_and_bigquery.py:89} INFO - Saved golden data to /tmp/tmp9ns70jfj/golden_data_20250502_105233.csv
[2025-05-02T10:53:02.713+0000] {gcs.py:560} INFO - File /tmp/tmp9ns70jfj/golden_data_20250502_105233.csv uploaded to golden_layer/golden_data_20250502_105233.csv in mysql-to-gcs-bronze-layer bucket
[2025-05-02T10:53:02.715+0000] {gcs_silver_to_gold_layer_and_bigquery.py:97} INFO - Uploaded golden_data_20250502_105233.csv to golden_layer/
[2025-05-02T10:53:02.748+0000] {connection.py:274} WARNING - Connection schemes (type: google_cloud_platform) shall not contain '_' according to RFC3986.
[2025-05-02T10:53:02.792+0000] {base.py:84} INFO - Using connection ID 'google_cloud_default' for task execution.
[2025-05-02T10:53:04.663+0000] {gcs_silver_to_gold_layer_and_bigquery.py:116} INFO - Dataset golden_layer already exists.
[2025-05-02T10:53:04.671+0000] {gcs_silver_to_gold_layer_and_bigquery.py:159} ERROR - Error during transformation or upload: BigQueryHook.run_load() got an unexpected keyword argument 'job_config'
Traceback (most recent call last):
  File "/opt/airflow/dags/gcs_silver_to_gold_layer_and_bigquery.py", line 151, in silver_to_golden
    bq_hook.run_load(
  File "/home/airflow/.local/lib/python3.12/site-packages/deprecated/classic.py", line 285, in wrapper_function
    return wrapped_(*args_, **kwargs_)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: BigQueryHook.run_load() got an unexpected keyword argument 'job_config'
[2025-05-02T10:53:04.682+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-05-02T10:53:04.684+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/gcs_silver_to_gold_layer_and_bigquery.py", line 151, in silver_to_golden
    bq_hook.run_load(
  File "/home/airflow/.local/lib/python3.12/site-packages/deprecated/classic.py", line 285, in wrapper_function
    return wrapped_(*args_, **kwargs_)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: BigQueryHook.run_load() got an unexpected keyword argument 'job_config'
[2025-05-02T10:53:04.842+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=gcs_silver_to_golden_bigquery, task_id=transform_and_upload_golden, run_id=manual__2025-05-02T10:52:33.516872+00:00, execution_date=20250502T105233, start_date=20250502T105252, end_date=20250502T105304
[2025-05-02T10:53:05.093+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 60 for task transform_and_upload_golden (BigQueryHook.run_load() got an unexpected keyword argument 'job_config'; 2308)
[2025-05-02T10:53:05.314+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-05-02T10:53:05.388+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-02T10:53:05.505+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
